docs/gc.md

```diff
@@ -141,72 +141,90 @@ Object ages are tracked for each region with a maximum of 24 possible generation
 
-Over time, transient objects drop out of the live set and viable survivor regions become scarce. This problem is solved by introducing a global mark phase (GMP) to mark and sweep the entire heap. GMP is a hybrid *incremental-concurrent* GC cycle. Each GMP increment is interleaved with a PGC cycle so that regions that are left sparse after sweeping the heap can be collected, cleared, and recycled during and after the GMP.
+Partial GC cycles work to reclaim free regions in the heap for allocating new objects. Because some objects from *eden* regions always survive, a partial GC cycle can reclaim only about  90% of this memory. To keep up with object allocation, partial GC cycles also reclaim free regions by defragmenting older regions. For example, a partial GC cycle that moves objects from 5 fragmented older regions into 2 empty regions, reclaims 3 regions for new object allocation. However, over time the overall amount of fragmented memory decreases and records about object liveness in older regions becomes less accurate. Eventually, the work done by partial GC cycles to reclaim memory cannot keep pace with memory allocation. Free regions become so scarce that a global *mark* operation (GMP) is required to build a new record of object liveness across the heap. A *sweep* operation uses this record to measure the amount of free memory in fragmented older regions, which later partial GC cycles can act upon to move objects and reclaim free regions.
 
-With the `balanced` policy, a global GC cycle is rare, occurring only in very tight memory conditions when PGC and GMP operations cannot free enough memory on the heap. These global GCs are explicit GCs, like those invoked by an application calling `System.gc()`.
+A global *sweep* operation also runs to reclaim memory so that it can create empty regions. The global *sweep* operation, while logically associated with the global *mark* operation, runs in the first partial GC cycle after the *mark* operation completes. Because the GC cycle responsible for the global *mark* operation runs concurrently, it might overlap and interleave with a few partial GC cycles.
```

Comment:
instead of 'runs in the first partial GC cycle', this is more correct to say (what may help with your confusion, but then may introduce some new ones?):

runs within same STW increment as first partial GC (cycle)